pkcs11/src/backend/db/object.rs

```diff
@@ -415,8 +416,12 @@ pub fn from_cert_data(
     cert: Vec<u8>,
     key_id: &str,
     raw_id: Option<Vec<u8>>,
+    certificate_format: CertificateFormat,
 ) -> Result<Object, Error> {
-    let cert = x509_cert::Certificate::from_pem(cert).map_err(Error::Der)?;
+    let cert = match certificate_format {
+        CertificateFormat::Pem => x509_cert::Certificate::from_pem(&cert).map_err(Error::Der)?,
+        CertificateFormat::Der => x509_cert::Certificate::from_der(&cert).map_err(Error::Der)?,
+    };
```

Comment:
- Please add a comment why we parse and re-encode the DER data (I assume for validation?).
- Maybe add a log message that mentions the configuration option?